File: app/src/main/java/com/example/friendlocation/friendlocation/JavaClasses/Model/Notification.java
package com.example.friendlocation.friendlocation.JavaClasses.Model;

import java.util.Map;

public class Notification {
    String type;
    int meetup_id;
    int organizer_id;
    int user_id;
    boolean new_status;

    public Notification(String type, int meetup_id, int organizer_id) {
        this.type = type;
        this.meetup_id = meetup_id;
        this.organizer_id = organizer_id;
    }

    public Notification(String type, int meetup_id, int user_id, boolean new_status) {
        this.type = type;
        this.meetup_id = meetup_id;
        this.user_id = user_id;
        this.new_status = new_status;
    }

    public Notification(String type, int meetup_id, boolean new_status) {
        this.type = type;
        this.meetup_id = meetup_id;
        this.new_status = new_status;
    }

    public Notification(Map<String, String> map) {
        this.type = map.get("type");
        this.meetup_id = Integer.valueOf(map.get("meetup_id"));
        this.organizer_id = Integer.valueOf(map.get("organizer_id"));
        this.user_id = Integer.valueOf(map.get("user_id"));
        this.new_status = Boolean.valueOf(map.get("new_status"));
        //Change to kotlin it will be better.
        //CheckNull // Not implemented
    }

    String getMeetupName(){
        return null; // Not implemented
    }

    String getOrganizerName(){
        return null; // Not implemented
    }

}

File: app/src/main/java/com/example/friendlocation/friendlocation/JavaClasses/Model/Notification.kt
package com.example.friendlocation.friendlocation.JavaClasses.Model

class Notification {
    var type: String?
        internal set
    var meetup_id: String?
        internal set
    var organizer_id: String?
        internal set
    var user_id: String? = ""
        internal set
    var isNew_status: Boolean? = false
        internal set

    constructor(type: String, meetup_id: String, organizer_id: String) {
        this.type = type
        this.meetup_id = meetup_id
        this.organizer_id = organizer_id
    }


    constructor(map: Map<String, String>) {
        this.type = map["type"]
        this.meetup_id = map["meetup_id"]
        this.organizer_id = map["organizer_id"]
        this.user_id = map["user_id"]
        this.isNew_status = java.lang.Boolean.valueOf(map["new_status"])
        //Change to kotlin it will be better.
        //CheckNull // Not implemented
    }
}
